src/main/java/org/zalando/nakadi/service/subscription/state/StartingState.java

```diff
@@ -54,15 +58,28 @@ public void onEnter() {
      * <p>
      * 4. Switches to streaming state.
      */
-    private void createSubscriptionLocked() {
+    private void initializeStream() {
         final boolean subscriptionJustInitialized = initializeSubscriptionLocked(getZk(),
                 getContext().getSubscription(), getContext().getTimelineService(), getContext().getCursorConverter());
         if (!subscriptionJustInitialized) {
+            // check if amount of streams <= the total amount of partitions
+            final Collection<Session> sessions = getZk().listSessions();
             final Partition[] partitions = getZk().getTopology().getPartitions();
             if (getZk().listSessions().size() >= partitions.length) {
```

Comment:
You can reuse `sessions` here instead of calling `getZk().listSessions()` a second time

Reply:
see line 66

Reply:
good catch; fixed https://github.com/zalando/nakadi/pull/848/commits/74026cef64c5118980542664e8d6c5b7bf25e316